libyul/optimiser/ForLoopConditionIntoBody.cpp

```diff
@@ -54,3 +56,45 @@ void ForLoopConditionIntoBody::operator()(ForLoop& _forLoop)
 	ASTModifier::operator()(_forLoop);
 }
 
+void ForLoopConditionOutOfBody::operator()(ForLoop& _forLoop)
+{
+	ASTModifier::operator()(_forLoop);
+
+	if (
+		!m_dialect.booleanNegationFunction() ||
+		_forLoop.condition->type() != typeid(Literal) ||
+		valueOfLiteral(boost::get<Literal>(*_forLoop.condition)) == u256(0) ||
+		_forLoop.body.statements.empty() ||
+		_forLoop.body.statements.front().type() != typeid(If)
+	)
+		return;
+
+	If& firstStatement = boost::get<If>(_forLoop.body.statements.front());
+	if (
+		firstStatement.body.statements.size() != 1 ||
```

Comment:
That could actually be ``firstStatement.body.statements.empty()`` but doesn't really matter.